diff --git a/cmake/FindSalomePythonLibs.cmake b/cmake/FindSalomePythonLibs.cmake
index 4ef691a..b246cca 100644
--- a/cmake/FindSalomePythonLibs.cmake
+++ b/cmake/FindSalomePythonLibs.cmake
@@ -35,7 +35,7 @@ IF (SALOMEPYTHONINTERP_FOUND AND NOT "${PYTHON_VERSION_STRING}" STREQUAL "")
    SET(PythonLibs_FIND_VERSION ${PYTHON_VERSION_STRING})
    SET(PythonLibs_FIND_VERSION_MAJOR ${PYTHON_VERSION_MAJOR})
 ENDIF()
-IF(WIN32)
+IF(WIN32 AND NOT MINGW)
   SET(CMAKE_LIBRARY_PATH "${PYTHON_ROOT_DIR}/libs")
   SET(PYTHON_INCLUDE_DIR "${PYTHON_ROOT_DIR}/include")
 ENDIF()
