diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index b157cd56e..02ba8e536 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -108,11 +108,13 @@ MACRO(SET_INSTALL_RPATHS _target _paths)
 ENDMACRO()
 
 # Find Boost
-IF(WIN32)
-	SET(Boost_USE_STATIC_LIBS ON)
-	SET(Boost_USE_STATIC_RUNTIME ON)
-	SET(Boost_USE_MULTITHREADED ON)
-ENDIF()
+# Do not force ON here, the proper options will be set on cmake
+# command line.
+#IF(WIN32)
+#	SET(Boost_USE_STATIC_LIBS ON)
+#	SET(Boost_USE_STATIC_RUNTIME ON)
+#	SET(Boost_USE_MULTITHREADED ON)
+#ENDIF()
 
 set(BOOST_COMPONENTS system program_options regex thread date_time)
 if(USE_MMAP)
