diff -Naur incertitudes.otgui-phimeca/CMakeLists.txt incertitudes.otgui-phimeca_correction/CMakeLists.txt
--- incertitudes.otgui-phimeca/CMakeLists.txt	2021-12-17 14:44:46.535138748 +0100
+++ incertitudes.otgui-phimeca_correction/CMakeLists.txt	2021-12-17 14:47:56.984036817 +0100
@@ -124,6 +124,12 @@
   endif ()
 endif ()
 
+find_package (Boost COMPONENTS filesystem locale REQUIRED)
+if (TARGET Boost::filesystem)
+  set (Boost_LIBRARIES Boost::filesystem Boost::locale)
+endif ()
+include_directories (${Boost_INCLUDE_DIRS})
+
 if (USE_OTFMI)
   include (FindPythonModule)
   find_python_module (otfmi)
