diff --git a/CMakeLists.txt b/CMakeLists.txt
index 116177dd..199c5f45 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -159,7 +159,7 @@ if (BUILD_PYTHON)
     endif ()
     if (SPHINX_EXECUTABLE AND (DVISVGM_EXECUTABLE OR DVIPNG_EXECUTABLE))
       find_python_module (sphinx 1.8)
-      find_python_module (numpydoc 0.9)
+      find_python_module (numpydoc 0.7)
     endif ()
     if (NOT SPHINX_EXECUTABLE OR NOT SPHINX_FOUND OR NOT NUMPYDOC_FOUND)
       set (SPHINX_FOUND FALSE)
