diff -Naur rkcommon-1.5.1/CMakeLists.txt rkcommon-1.5.1.dev/CMakeLists.txt
--- rkcommon-1.5.1/CMakeLists.txt	2020-09-24 17:19:32.000000000 +0200
+++ rkcommon-1.5.1.dev/CMakeLists.txt	2024-07-31 11:30:26.510662600 +0200
@@ -9,7 +9,7 @@
   cmake_minimum_required(VERSION 3.1)
 endif()
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
diff -Naur rkcommon-1.5.1/rkcommon/math/constants.h rkcommon-1.5.1.dev/rkcommon/math/constants.h
--- rkcommon-1.5.1/rkcommon/math/constants.h	2020-09-24 17:19:32.000000000 +0200
+++ rkcommon-1.5.1.dev/rkcommon/math/constants.h	2024-07-31 13:24:41.849946900 +0200
@@ -12,6 +12,26 @@
 #include <math.h>  // using cmath causes issues under Windows
 #include <cfloat>
 
+#if defined __MINGW32__
+#if !defined _MATH_DEFINES_DEFINED && !defined M_PI
+  #define _MATH_DEFINES_DEFINED
+
+  #define M_E        2.71828182845904523536   // e
+  #define M_LOG2E    1.44269504088896340736   // log2(e)
+  #define M_LOG10E   0.434294481903251827651  // log10(e)
+  #define M_LN2      0.693147180559945309417  // ln(2)
+  #define M_LN10     2.30258509299404568402   // ln(10)
+  #define M_PI       3.14159265358979323846   // pi
+  #define M_PI_2     1.57079632679489661923   // pi/2
+  #define M_PI_4     0.785398163397448309616  // pi/4
+  #define M_1_PI     0.318309886183790671538  // 1/pi
+  #define M_2_PI     0.636619772367581343076  // 2/pi
+  #define M_2_SQRTPI 1.12837916709551257390   // 2/sqrt(pi)
+  #define M_SQRT2    1.41421356237309504880   // sqrt(2)
+  #define M_SQRT1_2  0.707106781186547524401  // 1/sqrt(2)
+#endif
+#endif
+
 namespace rkcommon {
   namespace math {
 
diff -Naur rkcommon-1.5.1/rkcommon/math/rkmath.h rkcommon-1.5.1.dev/rkcommon/math/rkmath.h
--- rkcommon-1.5.1/rkcommon/math/rkmath.h	2020-09-24 17:19:32.000000000 +0200
+++ rkcommon-1.5.1.dev/rkcommon/math/rkmath.h	2024-07-31 12:59:28.820039300 +0200
@@ -9,7 +9,7 @@
 #include <algorithm>  // std::min()/std::max() on Windows
 #include <cmath>
 
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__)
 #include <intrin.h>
 #if (__MSV_VER <= 1700)
 namespace std {
diff -Naur rkcommon-1.5.1/rkcommon/networking/Socket.cpp rkcommon-1.5.1.dev/rkcommon/networking/Socket.cpp
--- rkcommon-1.5.1/rkcommon/networking/Socket.cpp	2020-09-24 17:19:32.000000000 +0200
+++ rkcommon-1.5.1.dev/rkcommon/networking/Socket.cpp	2024-07-31 11:30:04.130122300 +0200
@@ -52,7 +52,7 @@
 
   void initialize()
   {
-    static std::atomic<bool> initialized = false;
+    static std::atomic<bool> initialized = {false};
     static std::mutex initMutex;
     if (!initialized) {
       std::lock_guard<std::mutex> lock(initMutex);
diff -Naur rkcommon-1.5.1/rkcommon/networking/Socket.h rkcommon-1.5.1.dev/rkcommon/networking/Socket.h
--- rkcommon-1.5.1/rkcommon/networking/Socket.h	2020-09-24 17:19:32.000000000 +0200
+++ rkcommon-1.5.1.dev/rkcommon/networking/Socket.h	2024-07-31 11:00:21.438853900 +0200
@@ -16,7 +16,7 @@
   /*! exception thrown when other side disconnects */
   struct Disconnect : public std::exception
   {
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__)
     virtual const char *what() const override
 #else
     virtual const char *what() const noexcept override
diff -Naur rkcommon-1.5.1/rkcommon/os/library.cpp rkcommon-1.5.1.dev/rkcommon/os/library.cpp
--- rkcommon-1.5.1/rkcommon/os/library.cpp	2020-09-24 17:19:32.000000000 +0200
+++ rkcommon-1.5.1.dev/rkcommon/os/library.cpp	2024-07-31 12:55:22.288297400 +0200
@@ -36,7 +36,7 @@
   {
 #if defined(_WIN32) && !defined(__CYGWIN__)
     MEMORY_BASIC_INFORMATION mbi;
-    VirtualQuery(&_rkcommon_anchor, &mbi, sizeof(mbi));
+    VirtualQuery((LPCVOID)&_rkcommon_anchor, &mbi, sizeof(mbi));
     char pathBuf[16384];
     if (!GetModuleFileNameA(
             static_cast<HMODULE>(mbi.AllocationBase), pathBuf, sizeof(pathBuf)))
@@ -136,7 +136,7 @@
   void *Library::getSymbol(const std::string &sym) const
   {
 #ifdef _WIN32
-    return GetProcAddress((HMODULE)lib, sym.c_str());
+    return (void*)GetProcAddress((HMODULE)lib, sym.c_str());
 #else
     return dlsym(lib, sym.c_str());
 #endif
diff -Naur rkcommon-1.5.1/tests/CMakeLists.txt rkcommon-1.5.1.dev/tests/CMakeLists.txt
--- rkcommon-1.5.1/tests/CMakeLists.txt	2020-09-24 17:19:32.000000000 +0200
+++ rkcommon-1.5.1.dev/tests/CMakeLists.txt	2024-07-31 13:04:23.198162100 +0200
@@ -22,7 +22,7 @@
   memory/test_malloc.cpp
   memory/test_RefCount.cpp
 
-  os/test_FileName.cpp
+  #os/test_FileName.cpp
   os/test_library.cpp
 
   containers/test_AlignedVector.cpp
