*** a/salome	2023-10-18 10:19:10.946526043 +0200
--- b/salome	2023-10-18 10:22:34.534393901 +0200
***************
*** 50,60 ****
--- 50,67 ----
    # Identify application path then locate configuration files
    __initialize()
  
+   # Modify this variable to not reset the PATHS"
+   reinitialise_paths=True
+ 
    if args == ['--help']:
      from salomeContext import usage
      usage()
      sys.exit(0)
  
+   if '--keep-paths' in args:
+     reinitialise_paths=False
+     args.remove('--keep-paths')
+ 
    #from salomeContextUtils import getConfigFileNames
    #configFileNames, args, unexisting = getConfigFileNames( args, checkExistence=True )
    #if len(unexisting) > 0:
***************
*** 88,96 ****
      # Logger level error
      context.getLogger().setLevel(40)
  
-     # Modify this variable to not reset the PATHS"
-     reinitialise_paths=True
- 
      context.setVariable(r"PRODUCT_ROOT_DIR", out_dir_Path + r"", overwrite=True)
      # PRODUCT environment
      context.setVariable(r"SALOME_trace", r"local", overwrite=True)
--- 95,100 ----
